app/core/Tracker2/SegmentTracker.js

```diff
@@ -94,6 +94,17 @@ export function loadSegment () {
   // analytics.page();
 }
 
+const DEFAULT_SEGMENT_OPTIONS = {
+  // TODO remove after intercom disabled in segment
+  Intercom: { hideDefaultLauncher: true }
+}
+
+const DEFAULT_SEGMENT_TRAITS_TO_REPORT = [
```

Comment:
Massive nitpick: [94.27%](https://caniuse.com/#feat=mdn-javascript_builtins_set_set_null_allowed) of users globally should have the `Set` object that has quicker lookup. If we use a Set we could reduce the linear scan complexity on `includes` later: https://github.com/codecombat/codecombat/pull/5797/files#diff-9fb1ae5bb7b19f87f95e7c230fc75b0eR165

Reply:
Good call out!  Looking at this in the context of this comment made me realize that the way I'm filtering attributes is a bit backwards right now.  Instead of looping over the me attributes and then checking for the existence of the attribute in this array I can just loop over this array and check for the existence of the attribute on the me object.  Makes the code far simpler and keeps us at 100% browser coverage.  We can switch to a set if / when the attributes that are sent become more dynamic (allowing us to dedupe attributes).

Reply:
Thoughts?

Reply:
I like that! Iterating over the traits is a great idea!